src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts

```diff
@@ -243,6 +247,21 @@ export class BuyCrypto extends IEntity {
   // --- ENTITY METHODS --- //
 
   calculateOutputReferenceAmount(price: Price): this {
+    if (
+      Config.exchangeRateFromLiquidityOrder.includes(this.outputAsset.name) &&
+      this.liquidityPipeline?.orders?.length // TODO more checks needed? Is order always filled with exchangePrice at this point?
+    ) {
```

Comment:
If the pipeline is still running, we cannot calculate the output amount. Would need to abort here 🤔  if it's failed just use the normal pricing, if it's successful, the price should be there.